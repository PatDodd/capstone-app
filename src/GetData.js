import React, { Component } from 'react';

import { Container, Card, CardActionArea, CardActions, CardContent, Typography, Button } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { Timeline, OpenInNew, BarChart, TableChart, Map, Room, ShowChart} from '@material-ui/icons';

import { Link } from '@reach/router';

 
const styles = theme => ({
    card: {
        width: '25%',
        maxHeight: '100%',
        backgroundColor: theme.palette.secondary.main,
        color: 'white',
        marginTop: '20px',
        paddingLeft: '20px',
        marginRight: '20px',
        display: 'inline-block',
        [theme.breakpoints.up('lg')]: {
            width: '25%',
        },
        [theme.breakpoints.down('md')]: {
            width: '29%',
        },
        [theme.breakpoints.down('sm')]: {
            width: '48%',
        },
        [theme.breakpoints.down('xs')]: {
            width: '90%',
            margin: 'auto',
            display: 'block',
            marginTop: '10px'
        },
        
    },
    lineplotIcon: {
        transform: 'translate(0px, 5px)'
    },
    button: {
        color: 'white',
        backgroundColor: theme.palette.primary.light,
        '&:hover': {
            backgroundColor: theme.palette.primary.contrastText
        },
        padding: '10px'
    },
    link: {
        color: 'white'
    },
    openIcon: {
        transform: 'translate(0px, -1px)',
        fontSize: '1.3em'
    }
});

class GetData extends Component {
    
    constructor(props){
        
        super(props);
        
        this.state = {
            
        }        
    }

    render(){
        
        const { classes } = this.props;

        return(
            <Container>
                <Card className={classes.card}>
                    <CardActionArea>
                        <CardContent>
                            <Typography gutterBottom variant="body1" component="p">
                                Average Pace - Line Plot <Timeline className={classes.lineplotIcon} />
                            </Typography>
                        </CardContent>
                    </CardActionArea>
                    <CardActions>
                        <Link className={classes.link} to="/getdata/pace">
                            <Button size="medium" className={classes.button}>
                                OPEN CHART TYPE &nbsp; <OpenInNew className={classes.openIcon} />
                            </Button>
                        </Link>
                    </CardActions>
                </Card>
                <Card className={classes.card}>
                    <CardActionArea>
                        <CardContent>
                            <Typography gutterBottom variant="body1" component="p">
                                Effort Breakdown - Bar Chart <BarChart className={classes.lineplotIcon} />
                            </Typography>
                        </CardContent>
                    </CardActionArea>
                    <CardActions>
                    <Link className={classes.link} to="/getdata/effortpercent">
                        <Button size="medium" className={classes.button}>
                                OPEN CHART TYPE &nbsp; <OpenInNew className={classes.openIcon} />
                        </Button>
                    </Link>
                    </CardActions>
                </Card>
                <Card className={classes.card}>
                    <CardActionArea>
                        <CardContent>
                            <Typography gutterBottom variant="body1" component="p">
                                Stats - Table &amp; Graph &nbsp;
                                <TableChart className={classes.lineplotIcon} />
                                <BarChart className={classes.lineplotIcon} />
                            </Typography>
                        </CardContent>
                    </CardActionArea>
                    <CardActions>
                    <Link className={classes.link} to="/getdata/stats">
                        <Button size="medium" className={classes.button}>
                                OPEN CHART TYPE &nbsp; <OpenInNew className={classes.openIcon} />
                        </Button>
                    </Link>
                    </CardActions>
                </Card>
                <Card className={classes.card}>
                    <CardActionArea>
                        <CardContent>
                            <Typography gutterBottom variant="body1" component="p">
                                 Heatmap of my Runs - Map &nbsp;
                                <Map className={classes.lineplotIcon} />
                                <Room className={classes.lineplotIcon} />
                            </Typography>
                        </CardContent>
                    </CardActionArea>
                    <CardActions>
                    <Link className={classes.link} to="/getdata/heatmap">
                        <Button size="medium" className={classes.button}>
                                OPEN CHART TYPE &nbsp; <OpenInNew className={classes.openIcon} />
                        </Button>
                    </Link>
                    </CardActions>
                </Card>
                <Card className={classes.card}>
                    <CardActionArea>
                        <CardContent>
                            <Typography gutterBottom variant="body1" component="p">
                                 Elevation Analysis - Line Graph &nbsp;
                                <ShowChart className={classes.lineplotIcon} />
                            </Typography>
                        </CardContent>
                    </CardActionArea>
                    <CardActions>
                    <Link className={classes.link} to="/getdata/elevation">
                        <Button size="medium" className={classes.button}>
                                OPEN CHART TYPE &nbsp; <OpenInNew className={classes.openIcon} />
                        </Button>
                    </Link>
                    </CardActions>
                </Card>
            </Container>
        );
    }
}



export default withStyles(styles)(GetData);