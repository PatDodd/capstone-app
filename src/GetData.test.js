import React from 'react';
import ReactDOM from 'react-dom';
import GetData from './GetData';
import renderer from 'react-test-renderer';

// smoke test
it('renders <GetData /> component without crashing, smoke test', () => {
  const div = document.createElement('div'); 
  ReactDOM.render(<GetData />, div);
});

// snapshot test
it('renders correctly on snapshot test', () => {
  const tree = renderer.create(<GetData />).toJSON();
  expect(tree).toMatchSnapshot();
});