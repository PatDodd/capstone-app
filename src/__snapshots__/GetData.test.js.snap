// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly on snapshot test 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root MuiCard-root GetData-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Average Pace - Line Plot 
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
            />
          </svg>
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        className="GetData-link-4"
        href="/getdata/pace"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text GetData-button-3"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            OPEN CHART TYPE   
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root GetData-openIcon-5"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root GetData-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Effort Breakdown - Bar Chart 
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 9.2h3V19H5zM10.6 5h2.8v14h-2.8zm5.6 8H19v6h-2.8z"
            />
          </svg>
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        className="GetData-link-4"
        href="/getdata/effortpercent"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text GetData-button-3"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            OPEN CHART TYPE   
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root GetData-openIcon-5"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root GetData-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Stats - Table & Graph  
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 10.02h5V21h-5zM17 21h3c1.1 0 2-.9 2-2v-9h-5v11zm3-18H5c-1.1 0-2 .9-2 2v3h19V5c0-1.1-.9-2-2-2zM3 19c0 1.1.9 2 2 2h3V10H3v9z"
            />
          </svg>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 9.2h3V19H5zM10.6 5h2.8v14h-2.8zm5.6 8H19v6h-2.8z"
            />
          </svg>
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        className="GetData-link-4"
        href="/getdata/stats"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text GetData-button-3"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            OPEN CHART TYPE   
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root GetData-openIcon-5"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root GetData-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Heatmap of my Runs - Map  
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.5 3l-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z"
            />
          </svg>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        className="GetData-link-4"
        href="/getdata/heatmap"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text GetData-button-3"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            OPEN CHART TYPE   
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root GetData-openIcon-5"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiCard-root GetData-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Elevation Analysis - Line Graph  
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root GetData-lineplotIcon-2"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
            />
          </svg>
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        className="GetData-link-4"
        href="/getdata/elevation"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text GetData-button-3"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            OPEN CHART TYPE   
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root GetData-openIcon-5"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </div>
</div>
`;
